pkg/kubelet/prober/prober.go

```diff
@@ -195,6 +200,24 @@ func (pb *prober) runProbe(probeType probeType, p *v1.Probe, pod *v1.Pod, status
 		klog.V(4).InfoS("TCP-Probe Host", "host", host, "port", port, "timeout", timeout)
 		return pb.tcp.Probe(host, port, timeout)
 	}
+
+	if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.GRPCContainerProbe) && p.GRPC != nil {
+		port, err := extractPort(intstr.IntOrString{
```

Comment:
so you use this method instead of just assignment from int for the `0 < port < 65536` check?

Reply:
As tcpsocket did in https://github.com/kubernetes/kubernetes/blob/1ef10462a48ed1d58054e855a07318cc38ebcd1a/pkg/kubelet/prober/prober.go#L205

Reply:
This seems pointless to me.  Validation already checked the port number.  This takes an int, converts to intstr, and then converts back to int.  At best it is redundant, but it forced me to go look at the body of `extractPort`.  I'd suggest NOT doing this anywhere, unless you actually want to handle named ports (which I don't suggest unless you have a really good reason).